tests/codecs/test_numpy.py

```diff
@@ -110,13 +120,59 @@ def test_encode_output(payload: np.ndarray, expected: ResponseOutput):
             RequestInput(name="foo", shape=[2], data=["foo", "bar"], datatype="BYTES"),
             np.array(["foo", "bar"], dtype=str),
         ),
+        (
+            RequestInput(name="foo", shape=[3], data=[1, 2, None], datatype="FP16"),
+            np.array([1, 2, np.NaN], dtype="float16"),
```

Comment:
nit: perhaps using float values here is a bit more appropriate because the dtype is float :thinking: just to make it more obvious

Reply:
Well, it's also a good test that what drives the type is the `datatype` field and not the format in which the values come (i.e. `2` vs `2.0`).

Reply:
You are right now that I think of it 👍